lib/sensors/i2c_converter.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <cstdint>
+#include <optional>
+
+#include <core/logger.hpp>
+#include <io/i2c.hpp>
+
+namespace hyped::sensors {
+
+// TODOLater: Implement this properly
+/**
+ * @brief ADC (Texas Instruments) to be used for the hall-effect sensors (Farnell)
+ * @details This class is used to convert the analog signal from the hall-effect sensors to a
+ * digital signal over I2c
+ */
+class I2cConverter {
```

Comment:
From description is there not a better name for this? I know it's a little bit pedantic but the name of this doesn't give you an intuitive sense of it's purpose in my opinion? As a non-sensors person though, I may very well be wrong...

Reply:
it could be `AnalogueToI2cConverter` but since we only use one, I shortened it (there really isn't much more to it, it literally just converts) (but also if we do come up with a better name, it could be done in the PR dealing with this sensor)